rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -357,14 +357,15 @@ private void savePendingFederation(ActivationConfig.ForBlock activations) {
         }
 
         byte[] serializedPendingFederation = serializePendingFederation(activations);
-        bridgeStorageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedPendingFederation);
 
         if (!activations.isActive(RSKIP123)) {
+            bridgeStorageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedPendingFederation);
             return;
         }
 
         // we only need to save the standard part of the fed since the emergency part is constant
         saveFederationFormatVersion(PENDING_FEDERATION_FORMAT_VERSION.getKey(), STANDARD_MULTISIG_FEDERATION.getFormatVersion());
+        bridgeStorageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedPendingFederation);
```

Comment:
This turned out a bit confusing, same call to saveToRepository repeated twice. The old way was clearer

```
if (activations.isActive(RSKIP123)) {
  saveFederationFormatVersion(PENDING_FEDERATION_FORMAT_VERSION.getKey(), STANDARD_MULTISIG_FEDERATION.getFormatVersion());
}

bridgeStorageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedPendingFederation);
```

By the way, `saveToRepository` should actually be `safeSaveToRepository` probably. There's no `saveToRepository` defined in StorageAcessor

Reply:
mmmm, i had in mind the discussion we had a couple of days ago about returning earlier when possible.


about the `saveToRepository` thing, you right, i messed up things trying to solve the error that appeared

Reply:
Yeah, but in this case I don't think it helps. Having the call repeated in 2 different flows is error prone. If you ever need to change it, you risk only changing it in one place

Reply:
ok, it makes sense
Actually we should have something like 

```
bridgeStorageAccessor.safeSaveToRepository(PENDING_FEDERATION_KEY.getKey(), 
serializedPendingFederation);

if (!activations.isActive(RSKIP123)) {
       return;
}

// we only need to save the standard part of the fed since the emergency part is constant
saveFederationFormatVersion(PENDING_FEDERATION_FORMAT_VERSION.getKey(), STANDARD_MULTISIG_FEDERATION.getFormatVersion());
```
but that order (saving first the serialized federation and then the format version) was causing a test error